plone/app/layout/viewlets/common.py

```diff
@@ -324,11 +326,14 @@ def navtree(self):
             'plone.types_use_view_action_in_listings', [])
 
         for brain in brains:
-            brain_path = '/'.join(brain.getPath().split('/'))
+            brain_path = brain.getPath()
             brain_parent_path = brain_path.rpartition('/')[0]
             if brain_parent_path == navtree_path:
                 # This should be already provided by the portal_tabs_view
                 continue
+            if not context_path.startswith(brain_path) and brain.exclude_from_nav:  # noqa: E501
```

Comment:
Can you switch this condition to first check `brain.exclude_from_nav`? That is a boolean, so the check will be slightly faster than the string comparison.

Reply:
done